### ISSUE TITLE
Test translation failing since forever since it runs in cron and check is visible nowhere.

### ISSUE THREAD

**Issue by Carreau** [napari/napari#3642]
See https://github.com/napari/napari/actions/workflows/test_translations.yml

I guess we could use https://github.com/marketplace/actions/create-an-issue or similar  for cron workflow to automatically open and update an issue when the cron fails.

**Comment by nclack** [comment]
fwiw, it looks like the problem is the `calframe[1][3]` reference on this line [here](https://github.com/napari/napari/blob/76bafdaec166ab47f1cff2ec474648a9bebc9428/napari/settings/__init__.py#L57).

**Comment by goanpeca** [comment]
>Test translation failing since forever since it runs in cron and check is visible nowhere.

It is kinda by design @Carreau, we basically do not bother users or contibutors with translations, but the maintainers fixes them on every release, so having that visibility would not help in the current workflow.

**Comment by Carreau** [comment]
> It is kinda by design @Carreau, we basically do not bother users or contibutors with translations, but the maintainers fixes them on every release, so having that visibility would not help in the current workflow.

Well I don't mean to make it visible in every PR – I agree this woudl be annoying, but I'd like something more obvious than "think about checking it every now and then". Hence why I suggest to auto open an issue when it starts failing.

**Comment by goanpeca** [comment]
> Hence why I suggest to auto open an issue when it starts failing.

Well as soon as there is a new release we will have a new issue open as soon as a PR "breaks" it, so we would go from very annoying to mildly annoying 🙃.

>"think about checking it every now and then"

It is actually more like check that workflow when preparing a napari release.

**Comment by nclack** [comment]
I may be misunderstanding something, but:

1. `test_strings.py::test_missing_translations` fails with an exception (hence my comment above). That looks like a bug to me?
2. even when 1 is fixed, napari fails the audit of missing strings. I guess this will be addressed before release.
3. it appears the visibility of the above issues has been low

Is there a way we can check that test_strings.py is working correctly (nothing introduced a bug) on each PR? 

fwiw, I do get emails about the test_translation failures for my fork. I just never look at them... I'd pay more attention if they were generating the output they were intended to have.

**Comment by goanpeca** [comment]
Hi @nclack 

>test_strings.py::test_missing_translations fails with an exception (hence my comment above). That looks like a bug to me?

Agreed, that might need a fix!

>even when 1 is fixed, napari fails the audit of missing strings. I guess this will be addressed before release.

Correct!

>it appears the visibility of the above issues has been low

I guess we could do a check that checks that things do not fail, so that we ensure we have correct code running, but do not fail in case of missing strings checks, since as stated before, we do that as part of the release process.

We could eventually start enforcing translations checks on PRs, we just decided against for now. As soon as we have a language pack available (or more), I would say we should probably move the checks to the PRs directly.

**Comment by Carreau** [comment]
I think having an issue automatically opened when there are missing translation strings is also good for first time contributions. 

1) It does not block the original PR author, 
2) It's automatic once the PR is merged. 
3) It's a well defined task that can be taken with someone w/o knowledge of the first PR
4) it's potentially less work for the release manager.

Even with many language pack, I think it would be ok to not put as an extra check in the PRs.

**Comment by goanpeca** [comment]
Sounds good @Carreau, thanks for bringing this up. I will add the checks on PRs just to make sure we do not have incorrect python code (so not failing if missing strings, but failing if an exception is raised while running tests :) )

Could you work on the other side of it :) ?

**Comment by goanpeca** [comment]
Updated the strings https://github.com/napari/napari/pull/3662/files

**Comment by jni** [comment]
Thanks both, I like the final decision here! :blush:

**Comment by psobolewskiPhD** [comment]
@brisvag with https://github.com/napari/napari/pull/8935 I guess this can be closed as irrelevant?
